src/libraries/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs

```diff
@@ -2866,6 +2866,46 @@ public async void PlaintextStreamFilter_CustomStream_Success(bool useSsl)
                     catch (IOException) { }
                 }, options: options);
         }
+
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public async Task PlaintextStreamFilter_Logging_Success(bool useSsl)
+        {
+            bool log = int.TryParse(Environment.GetEnvironmentVariable("DOTNET_TEST_SOCKETSHTTPHANDLERLOG"), out int value) && value == 1;
+
+            GenericLoopbackOptions options = new GenericLoopbackOptions() { UseSsl = useSsl };
+            await LoopbackServerFactory.CreateClientAndServerAsync(
+                async uri =>
+                {
+                    using HttpClientHandler handler = CreateHttpClientHandler();
+                    handler.ServerCertificateCustomValidationCallback = TestHelper.AllowAllCertificates;
+                    var socketsHandler = (SocketsHttpHandler)GetUnderlyingSocketsHttpHandler(handler);
+                    socketsHandler.PlaintextStreamFilter = (context, token) =>
+                    {
+                        Assert.Equal(HttpVersion.Version11, context.NegotiatedHttpVersion);
+
+                        bool foundGet = false;
+                        return ValueTask.FromResult<Stream>(new BytesLoggingStream(context.PlaintextStream, (read, data) =>
+                        {
+                            if (log)
+                            {
+                                Console.WriteLine(data);
+                            }
+
+                            foundGet |= read && data.Contains("GET / 1.1");
```

Comment:
Might be useful to validate the response stream as well, e.g. look for "200 OK" or "hello" from content body, or whatever/

Reply:
Also I assume you want to Assert.True(foundGet) here somewhere?

And I think the text you're looking for should be "GET / HTTP/1.1"

Reply:
Heh, yes.